packages/vue-app/template/index.js

```diff
@@ -246,16 +246,28 @@ async function createApp(ssrContext, config = {}) {
   // If server-side, wait for async component to be resolved first
   if (process.server && ssrContext && ssrContext.url) {
     await new Promise((resolve, reject) => {
-      router.push(ssrContext.url, resolve, () => {
-        // navigated to a different route in router guard
-        const unregister = router.afterEach(async (to, from, next) => {
-          ssrContext.url = to.fullPath
-          app.context.route = await getRouteData(to)
-          app.context.params = to.params || {}
-          app.context.query = to.query || {}
-          unregister()
+      router.push(ssrContext.url, resolve, (err) => {
+        if (!err._isRouter) {
+          // router guard returned an error
+          reject(err)
```

Comment:
we can do early return on this rejection `return reject(err)`

Reply:
done